### PR TITLE
Hide goals with state.type=completed,failed,canceled by default via filter

### LINKED ISSUE #1337: Hide goals with state.type=completed,failed,canceled by default via filter
Change header goals link with a global preset with this filter

Add an option "default" to a filter

### PR DESCRIPTION
**Pull request by sab94a**
## PR includes

- [x] Feature

## Related issues

Resolve #1337 

## QA Instructions, Screenshots, Recordings

https://github.com/taskany-inc/issues/assets/9020793/4426bc4c-994d-4d40-85f0-02a6c9fe9c64

### CODE REVIEW

**Comment by sab94a** [comment]
https://github.com/taskany-inc/issues/assets/9020793/bb280b2e-3b2b-4255-b966-a17a1ffcdc21

File: src/hooks/useUrlFilterParams.ts [hunk 1]
```diff
@@ -4,13 +4,17 @@ import { MouseEventHandler, useCallback, useMemo, useState } from 'react';
 import { Tag } from '@prisma/client';
 
 import { FilterById } from '../../trpc/inferredTypes';
-import { Priority } from '../types/priority';
 import { SortDirection, SortableProps } from '../components/SortFilter/SortFilter';
 
+import { useCookies } from './useCookies';
+
+export const filtersNoSearchPresetCookie = 'takany.NoSearchPreset';
```
Comment (awinogradov) [inline src/hooks/useUrlFilterParams.ts:11]: taskany

File: src/hooks/useUrlFilterParams.ts [hunk 2]
```diff
@@ -69,80 +128,41 @@ export const useUrlFilterParams = ({ preset }: { preset?: FilterById }) => {
...
+
+        return <T extends keyof QueryState>(key: T) =>
             (value: QueryState[T]) => {
-                setCurrentPreset(undefined);
-                pushNewState({
-                    ...queryState,
-                    [key]: value,
+                state[key] = value;
+
+                queueMicrotask(() => {
```
Comment (awinogradov) [inline src/hooks/useUrlFilterParams.ts:155]: ?

File: trpc/router/filter.ts [hunk 3]
```diff
@@ -7,21 +7,23 @@ import { connectionMap } from '../queries/connections';
 import { createFilterSchema } from '../../src/schema/filter';
 import { ToggleSubscriptionSchema } from '../../src/schema/common';
 
+const fitlerIncludeSchema = {
```
Comment (awinogradov) [inline trpc/router/filter.ts:10]: `includeStargizersAndActivity` or much better to separate it `includeStargizers`, `includeActivity`

File: trpc/router/project.ts [hunk 4]
```diff
@@ -275,8 +275,29 @@ export const project = router({
...
         )
-        .query(async ({ ctx, input: { goalsQuery } = {} }) => {
+        .query(async ({ ctx, input: { goalsQuery = {} } = {} }) => {
             const { activityId, role } = ctx.session.user;
+
+            const stateByTypes = goalsQuery?.stateType
+                ? await prisma.state.findMany({
+                      where: {
+                          type: {
+                              in: goalsQuery.stateType as StateType[],
```
Comment (awinogradov) [inline trpc/router/project.ts:285]: you can use prisma enum in zod schema to avoid `as` here

**Review by awinogradov — APPROVED** [review]

**Review by 9teen90nine — APPROVED** [review]

**Pull request merged**